src/main/java/com/thealgorithms/backtracking/MColoring.java

```diff
@@ -7,71 +7,91 @@
 import java.util.Set;
 
 /**
+ * Node class represents a graph node. Each node is associated with a color
+ * (initially 1) and contains a set of edges representing its adjacent nodes.
+ *
  * @author Bama Charan Chhandogi (https://github.com/BamaCharanChhandogi)
  */
 class Node {
-    int color = 1;
-    Set<Integer> edges = new HashSet<Integer>();
+    int color = 1; // Initial color for each node
+    Set<Integer> edges = new HashSet<Integer>(); // Set of edges representing adjacent nodes
 }
 
+/**
+ * MColoring class solves the M-Coloring problem where the goal is to determine
+ * if it's possible to color a graph using at most M colors such that no two
+ * adjacent nodes have the same color.
+ */
 public final class MColoring {
+
     private MColoring() {
-    }
+    } // Prevent instantiation of utility class
+
+    /**
+     * Determines whether it is possible to color the graph using at most M colors.
+     *
+     * @param nodes List of nodes representing the graph.
+     * @param n     The total number of nodes in the graph.
+     * @param m     The maximum number of allowed colors.
+     * @return 1 if the graph can be colored using M colors, otherwise 0.
+     */
     static int possiblePaint(ArrayList<Node> nodes, int n, int m) {
```

Comment:
Could you make it return boolean instead of int? And rename the function to `isColoringPossible`

Reply:
Done! I renamed the function in the test file as well.